test_automated_recovery_from_stopped_node_and_start[True] fails on VSPHERE6-IPI-IN-TRANSIT-ENCRYPTION deployments

Failed on ODF 4.18 as it is unable to surpass `get_node_rack_or_zone`

```
  node_names = set([node.name for node in nodes])
E   AttributeError: 'NoneType' object has no attribute 'name'



ocs_ci/ocs/platform_nodes.py:3020: AttributeError
```

RP- https://url.corp.redhat.com/734f7ce

Quoting Itzhak from our offline discussion:

```
I see in the node output https://url.corp.redhat.com/734d1b9 there are 2 worker nodes without rack. This is because the run was with 6 worker nodes. The function get_node_rack_or_zone returns ValueError when it doesn't find the node rack. This will not happen if we run with 3 worker nodes. Anyway, one fix is to add a param in the function get_node_rack_or_zone so it will not raise an exception when it doesn't find the rack or zone.
```

The teardown causes the error you mentioned in the issue. This is the error we need to fix: https://reportportal-ocs4.apps.dno.ocp-hub.prod.psi.redhat.com/ui/#ocs/launches/795/29655/1465680/1465698/log?logParams=history%3D1465697%26page.page%3D1.


```
@retry(ValueError, tries=5, delay=10)
def get_node_rack_or_zone(failure_domain, node_obj):
    """
    Get the worker node rack or zone name based on the failure domain value

    Args:
        failure_domain (str): The failure domain
        node_obj (ocs_ci.ocs.resources.ocs.OCS): The node object

    Returns:
        str: The worker node rack/zone name

    """
    node_obj.reload()
    node_rack_or_zone = (
        get_node_zone(node_obj) if failure_domain == "zone" else get_node_rack(node_obj)
    )
    if node_rack_or_zone:
        return node_rack_or_zone
    else:

      raise ValueError
E           ValueError
```

--- a/ocs_ci/ocs/node.py
+++ b/ocs_ci/ocs/node.py
@@ -2705,7 +2705,7 @@ def get_another_osd_node_in_same_rack_or_zone(
     return osd_node_in_same_rack_or_zone
 
 
-def get_nodes_racks_or_zones(failure_domain, node_names):
+def get_nodes_racks_or_zones(failure_domain, node_names, raise_value_error=True):
     """
     Get the nodes racks or zones
 
@@ -2717,10 +2717,15 @@ def get_nodes_racks_or_zones(failure_domain, node_names):
 
     """
     node_objects = get_node_objs(node_names)
-    return [get_node_rack_or_zone(failure_domain, n) for n in node_objects]
+    return [
+        get_node_rack_or_zone(failure_domain, n, raise_value_error)
+        for n in node_objects
+    ]
 
 
-def wait_for_nodes_racks_or_zones(failure_domain, node_names, timeout=120):
+def wait_for_nodes_racks_or_zones(
+    failure_domain, node_names, timeout=120, raise_value_error=True
+):
     """
     Wait for the nodes racks or zones to appear
 
@@ -2739,6 +2744,7 @@ def wait_for_nodes_racks_or_zones(failure_domain, node_names, timeout=120):
         func=get_nodes_racks_or_zones,
         failure_domain=failure_domain,
         node_names=node_names,
+        raise_value_error=raise_value_error,
     ):
         log.info(f"The nodes {node_names} racks or zones are: {nodes_racks_or_zones}")
         if all(nodes_racks_or_zones):

--- a/ocs_ci/ocs/node.py
+++ b/ocs_ci/ocs/node.py
@@ -2666,7 +2666,8 @@ def get_other_worker_nodes_in_same_rack_or_zone(
     other_wnodes_in_same_rack_or_zone = [
         wnode
         for wnode in other_wnodes
-        if get_node_rack_or_zone(failure_domain, wnode) == node_rack_or_zone
+        if get_node_rack_or_zone(failure_domain, wnode, raise_value_error)
+        == node_rack_or_zone
     ]
 
     wnode_names = [n.name for n in other_wnodes_in_same_rack_or_zone]

--- a/ocs_ci/ocs/node.py
+++ b/ocs_ci/ocs/node.py
@@ -2228,7 +2228,6 @@ def base_get_node_rack_or_zone(failure_domain, node_obj):
         raise ValueError(f"{failure_domain} was not found for the node {node_obj.name}")
 
 
-@retry(ValueError, tries=5, delay=10)
 def get_node_rack_or_zone(failure_domain, node_obj, raise_value_error=True):
     """
     Wrapper for base_get_node_rack_or_zone that allows optional suppression of ValueError.
@@ -2640,7 +2639,7 @@ def wait_for_all_osd_ids_come_up_on_nodes(
 
 
 def get_other_worker_nodes_in_same_rack_or_zone(
-    failure_domain, node_obj, node_names_to_search=None
+    failure_domain, node_obj, node_names_to_search=None, raise_value_error=True
 ):
     """
     Get other worker nodes in the same rack or zone of a given node.
@@ -2656,7 +2655,9 @@ def get_other_worker_nodes_in_same_rack_or_zone(
         list: The list of the other worker nodes in the same rack or zone of the given node.
 
     """
-    node_rack_or_zone = get_node_rack_or_zone(failure_domain, node_obj)
+    node_rack_or_zone = get_node_rack_or_zone(
+        failure_domain, node_obj, raise_value_error
+    )
     log.info(f"The node {node_obj.name} rack or zone is {node_rack_or_zone}")
     wnode_names = node_names_to_search or get_worker_nodes()
     other_wnode_names = [name for name in wnode_names if name != node_obj.name]
@@ -2674,7 +2675,7 @@ def get_other_worker_nodes_in_same_rack_or_zone(
 
 
 def get_another_osd_node_in_same_rack_or_zone(
-    failure_domain, node_obj, node_names_to_search=None
+    failure_domain, node_obj, node_names_to_search=None, raise_value_error=True
 ):
     """
     Get another osd node in the same rack or zone of a given node.
@@ -2693,7 +2694,7 @@ def get_another_osd_node_in_same_rack_or_zone(
     """
     osd_node_names = get_osd_running_nodes()
     other_wnodes_in_same_rack_or_zone = get_other_worker_nodes_in_same_rack_or_zone(
-        failure_domain, node_obj, node_names_to_search
+        failure_domain, node_obj, node_names_to_search, raise_value_error
     )
 
     osd_node_in_same_rack_or_zone = None

--- a/ocs_ci/ocs/node.py
+++ b/ocs_ci/ocs/node.py
@@ -2204,9 +2204,9 @@ def get_node_zone_dict():
 
 
 @retry(ValueError, tries=5, delay=10)
-def get_node_rack_or_zone(failure_domain, node_obj):
+def base_get_node_rack_or_zone(failure_domain, node_obj):
     """
-    Get the worker node rack or zone name based on the failure domain value
+    Get the worker node rack or zone name based on the failure domain value.
 
     Args:
         failure_domain (str): The failure domain
@@ -2215,6 +2215,8 @@ def get_node_rack_or_zone(failure_domain, node_obj):
     Returns:
         str: The worker node rack/zone name
 
+    Raises:
+        ValueError: If the zone/rack is not found after retries.
     """
     node_obj.reload()
     node_rack_or_zone = (
@@ -2223,7 +2225,29 @@ def get_node_rack_or_zone(failure_domain, node_obj):
     if node_rack_or_zone:
         return node_rack_or_zone
     else:
-        raise ValueError
+        raise ValueError(f"{failure_domain} was not found for the node {node_obj.name}")
+
+
+@retry(ValueError, tries=5, delay=10)
+def get_node_rack_or_zone(failure_domain, node_obj, raise_value_error=True):
+    """
+    Wrapper for base_get_node_rack_or_zone that allows optional suppression of ValueError.
+
+    Args:
+        failure_domain (str): The failure domain
+        node_obj (ocs_ci.ocs.resources.ocs.OCS): The node object
+        raise_value_error (bool): Whether to raise ValueError or return None on failure
+    Returns:
+        str or None: Rack/zone name or None if not found and raise_value_error is False
+
+    """
+    try:
+        return base_get_node_rack_or_zone(failure_domain, node_obj)
+    except ValueError as ex:
+        if raise_value_error:
+            raise ex
+        else:
+            return None
 
 
 def get_node_rack_or_zone_dict(failure_domain):

--- a/tests/functional/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py
+++ b/tests/functional/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py
@@ -325,7 +325,7 @@ class TestAutomatedRecoveryFromStoppedNodes(ManageTest):
             self.extra_node = True
             log.info("Get another OSD node in the same rack or zone...")
             self.osd_worker_node = get_another_osd_node_in_same_rack_or_zone(
-                failure_domain, new_ocs_node
+                failure_domain, new_ocs_node, raise_value_error=False
             )
             assert (
                 self.osd_worker_node

--- a/tests/functional/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py
+++ b/tests/functional/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py
@@ -317,8 +317,9 @@ class TestAutomatedRecoveryFromStoppedNodes(ManageTest):
             new_ocs_node_names = add_new_node_and_label_it(self.machineset_name)
             failure_domain = get_failure_domain()
             log.info("Wait for the nodes racks or zones to appear...")
-            wait_for_nodes_racks_or_zones(failure_domain, new_ocs_node_names)
-
+            wait_for_nodes_racks_or_zones(
+                failure_domain, new_ocs_node_names, raise_value_error=False
+            )
             new_ocs_node = get_node_objs(new_ocs_node_names)[0]
             log.info(f"Successfully created a new OCS node '{new_ocs_node.name}'")
             self.extra_node = True

